contracts/chain-adapters/Arbitrum_Adapter.sol

```diff
@@ -54,24 +57,36 @@ contract Arbitrum_Adapter is AdapterInterface, CircleCCTPAdapter {
     // Generic gateway: https://github.com/OffchainLabs/token-bridge-contracts/blob/main/contracts/tokenbridge/ethereum/gateway/L1ArbitrumGateway.sol
     ArbitrumL1ERC20GatewayLike public immutable L1_ERC20_GATEWAY_ROUTER;
 
+    // We don't allow fees to go higher than `OFT_FEE_CAP` when calling .send in `OFTTransportAdapter`. It gives us some protection from potential bugs on OFT's end
+    uint256 public constant OFT_FEE_CAP = 1 ether;
+
+    // Helper contract to help us map token -> OFT messenger for OFT-enabled tokens
+    OFTAddressBook public immutable OFT_ADDRESS_BOOK;
+
     /**
      * @notice Constructs new Adapter.
      * @param _l1ArbitrumInbox Inbox helper contract to send messages to Arbitrum.
      * @param _l1ERC20GatewayRouter ERC20 gateway router contract to send tokens to Arbitrum.
      * @param _l2RefundL2Address L2 address to receive gas refunds on after a message is relayed.
      * @param _l1Usdc USDC address on L1.
      * @param _cctpTokenMessenger TokenMessenger contract to bridge via CCTP.
+     * @param _oftAddressBook OFTAddressBook contract to helps identify token -> oftMessenger relationship for OFT bridging.
      */
     constructor(
         ArbitrumL1InboxLike _l1ArbitrumInbox,
         ArbitrumL1ERC20GatewayLike _l1ERC20GatewayRouter,
         address _l2RefundL2Address,
         IERC20 _l1Usdc,
-        ITokenMessenger _cctpTokenMessenger
-    ) CircleCCTPAdapter(_l1Usdc, _cctpTokenMessenger, CircleDomainIds.Arbitrum) {
+        ITokenMessenger _cctpTokenMessenger,
+        OFTAddressBook _oftAddressBook
+    )
+        CircleCCTPAdapter(_l1Usdc, _cctpTokenMessenger, CircleDomainIds.Arbitrum)
+        OFTTransportAdapter(OFTEIds.Arbitrum, OFT_FEE_CAP)
```

Comment:
I think its better to pas in this OFT_FEE_CAP as a constructor param and then update the deploy script to set this to 1 ether. Just in case we want to change it in the future then we wouldn't have to update the contract